pkg/broker/handler/handler.go

```diff
@@ -74,8 +75,17 @@ func (h *Handler) IsAlive() bool {
 func (h *Handler) receive(ctx context.Context, msg *pubsub.Message) {
 	ctx = metrics.StartEventProcessing(ctx)
 	event, err := binding.ToEvent(ctx, cepubsub.NewMessage(msg))
+	// The following errors can be returned by ToEvent and are not retryable.
+	// TODO Should ToEvent consolidate them and return the genric ErrCannotConvertToEvent?
+	if errors.Is(err, binding.ErrCannotConvertToEvent) || errors.Is(err, binding.ErrNotStructured) || errors.Is(err, binding.ErrUnknownEncoding) || errors.Is(err, binding.ErrNotBinary) {
+		logging.FromContext(ctx).Error("failed to convert received message to an event, check the msg format", zap.Any("message", msg), zap.Error(err))
+		// Ack the message so it won't be retried.
+		// TODO Should this go to the DLQ once DLQ is implemented?
+		//msg.Ack()
```

Comment:
Was this intended to be commented out?

Reply:
If we log the full message, then I think we should ack the message now. The reason being 1) retry such messages will continue to fail so it doesn't make sense to do so 2) since we log the message (with debug), developers can still check what's in the message that was causing the problem

Reply:
oops, sorry forgot to uncomment, was doing some testing.